cli/Valet/Site.php

```diff
@@ -145,6 +145,77 @@ function parked()
         return $parkedLinks;
     }
 
+    /**
+     * Get all sites which are proxies (not Links, and contain proxy_pass directive)
+     *
+     * @return \Illuminate\Support\Collection
+     */
+    function proxies()
+    {
+        $dir = VALET_HOME_PATH.'/Nginx/';
+        $tld = $this->config->read()['tld'];
+        $links = $this->links();
+        $certsPath = VALET_HOME_PATH.'/Certificates';
+        $certs = $this->getCertificates($certsPath);
+
+        $proxies = collect($this->files->scandir($dir))
+        ->filter(function ($site, $key) use ($tld) {
+            // keep sites that match our TLD
+            return ends_with($site, '.'.$tld);
+        })->map(function ($site, $key) use ($tld) {
+            // remove the TLD suffix for consistency
+            return str_replace('.'.$tld, '', $site);
+        })->reject(function ($site, $key) use ($links) {
+            return $links->has($site);
+        })->mapWithKeys(function ($site) {
+            $host = $this->getProxyHostForSite($site) ?: '(other)';
+            return [$site => $host];
+        })->reject(function ($host, $site) {
+            // if proxy host is null, it may be just a normal SSL stub, or something else; either way we exclude it from the list
+            return $host === '(other)';
+        })->map(function ($host, $site) use ($certs) {
+            $secured = $certs->has($site);
+            $url = ($secured ? 'https': 'http').'://'.$site;
+
+            return [
+                'site' => $site,
+                'secured' => $secured ? ' X': '',
+                'url' => $url,
+                'path' => $host,
+            ];
+        });
+
+        // $proxies = $proxies->merge($links);
+    
+        return $proxies;
+    }
+
+    /**
+     * Identify whether a site is for a proxy by reading the host name from its config file.
+     *
+     * @param string $site Site name without TLD
+     * @param string $configContents Config file contents
+     * @return string|null
+     */
+    function getProxyHostForSite($site, $configContents = null)
+    {
+        $siteConf = $configContents ?: $this->getSiteConfigFileContents($site);
+        $host = null;
+        if (preg_match('~proxy_pass (?<host>[^/]+/.*)~', $siteConf, $patterns)) {
```

Comment:
I've got two conflicting ideas on this.

Here are three commands I ran as I continued to see weird situations come up. Some of this is broken assumptions, of course, but it help me better understand things as I fine tuned the regular expression for various cases.

```
valet proxy foo bar
valet proxy foo2 /bar
valet proxy foo3 b/a/r
```

**First**, the current regular expression seems too strict/specific, and I'm not sure where it came from. It looks like it is saying, "the proxied host needs to not start with a leading `/`, but that it needs to have exactly one `/` at some point later." I don't know why this is a requirement? Could totally be just me not knowing how this might also hurt things, but it seems like we should trap *any* proxy pass host at this point.

```
+---------------------------+-----+-----------------------------------+--------------------------------------------------------------------------+
| Site                      | SSL | URL                               | Host                                                                     |
+---------------------------+-----+-----------------------------------+--------------------------------------------------------------------------+
| app-test.aaaaaaaaaaaa.com |  X  | https://app-test.aaaaaaaaaaaa.com | https://127.0.0.1:8443;                                                  |
| app.aaaaaaaaaaaa.com      |  X  | https://app.aaaaaaaaaaaa.com      | https://127.0.0.1:8443;                                                  |
| foo                       |  X  | https://foo                       | bar;                                                                     |
|                           |     |                                   |         proxy_set_header   Host              $host;                      |
|                           |     |                                   |         proxy_set_header   X-Real-IP         $remote_addr;               |
|                           |     |                                   |         proxy_set_header   X-Forwarded-For   $proxy_add_x_forwarded_for; |
|                           |     |                                   |         proxy_set_header   X-Forwarded-Proto $scheme;                    |
|                           |     |                                   |         proxy_set_header   X-Client-Verify   SUCCESS;                    |
|                           |     |                                   |         proxy_set_header   X-Client-DN       $ssl_client_s_dn;           |
|                           |     |                                   |         proxy_set_header   X-SSL-Subject     $ssl_client_s_dn;           |
|                           |     |                                   |         proxy_set_header   X-SSL-Issuer      $ssl_client_i_dn;           |
|                           |     |                                   |         proxy_set_header   X-NginX-Proxy true;                           |
|                           |     |                                   |         proxy_set_header   Upgrade $http_upgrade;                        |
|                           |     |                                   |         proxy_set_header   Connection "upgrade";                         |
|                           |     |                                   |         proxy_http_version 1.1;                                          |
|                           |     |                                   |         proxy_read_timeout 1800;                                         |
|                           |     |                                   |         proxy_connect_timeout 1800;                                      |
|                           |     |                                   |         chunked_transfer_encoding on;                                    |
|                           |     |                                   |         proxy_redirect off;                                              |
|                           |     |                                   |         proxy_buffering off;                                             |
|                           |     |                                   |     }                                                                    |
|                           |     |                                   |                                                                          |
|                           |     |                                   |     location ~ /\.ht {                                                   |
| foo3                      |  X  | https://foo3                      | b/a/r;                                                                   |
+---------------------------+-----+-----------------------------------+--------------------------------------------------------------------------+
```

As it stands right now, it shows `foo` and `foo3`, but not `foo2`. The value captured for `foo` is not what is expected.

At the very least, I'd suggest we terminate the regular expression by not capturing any whitespace before the trailing `;` which I believe is required to terminate the Nginx config line anyway.

So, my naive solution would be to use the following regex instead: `~proxy_pass\s+(?<host>.*)\s*;~`

By using this, I was able to see all of the proxies I added using the `valet proxy` command and shows the value we might expect:

```
+---------------------------+-----+-----------------------------------+------------------------+
| Site                      | SSL | URL                               | Host                   |
+---------------------------+-----+-----------------------------------+------------------------+
| app-test.aaaaaaaaaaaa.com |  X  | https://app-test.aaaaaaaaaaaa.com | https://127.0.0.1:8443 |
| app.aaaaaaaaaaaa.com      |  X  | https://app.aaaaaaaaaaaa.com      | https://127.0.0.1:8443 |
| foo                       |  X  | https://foo                       | bar                    |
| foo2                      |  X  | https://foo2                      | /bar                   |
| foo3                      |  X  | https://foo3                      | b/a/r                  |
+---------------------------+-----+-----------------------------------+------------------------+
```

**Second**, none of these `foo`, `foo2`, `foo3` proxies actually work! Or rather, they add the configs, but breaks Nginx in the process:

```
$ valet proxy foo bar
Valet will now proxy [https://foo.test] traffic to [bar].

In Nginx.php line 121:

  Nginx cannot start; please check your nginx.conf [1: nginx: [emerg] invalid URL prefix in /User
  s/simensen/.config/valet/Nginx/foo.test:35
  nginx: configuration file /usr/local/etc/nginx/nginx.conf test failed
  ].


proxy <domain> <host>
```
```
$ valet proxy foo2 /bar
Valet will now proxy [https://foo2.test] traffic to [/bar].

In Nginx.php line 121:

  Nginx cannot start; please check your nginx.conf [1: nginx: [emerg] invalid URL prefix in /User
  s/simensen/.config/valet/Nginx/foo2.test:35
  nginx: configuration file /usr/local/etc/nginx/nginx.conf test failed
  ].


proxy <domain> <host>
```
```
$ valet proxy foo3 b/a/r
Valet will now proxy [https://foo3.test] traffic to [b/a/r].

In Nginx.php line 121:

  Nginx cannot start; please check your nginx.conf [1: nginx: [emerg] invalid URL prefix in /User
  s/simensen/.config/valet/Nginx/foo3.test:35
  nginx: configuration file /usr/local/etc/nginx/nginx.conf test failed
  ].


proxy <domain> <host>
```

So, I think my regex is better (`~proxy_pass\s+(?<host>.*)\s*;~`) if only because it catches all of the possibilities for proxies added with `valet proxy domain host` and because it lists them correctly (_and_ without the trailing `;`). However, I don't know if there are other reasons for the original regex since it was quite specific. My use cases could be too simple/naive.

I think this also means we should actually be checking the input for the proxy host value for Nginx to work, anyway, since running `valet proxy domain url` should not break Nginx. If we do that, we could be more specific about the regular expression for proxies to match http(s):// urls. For that, I came up with this solution: `~proxy_pass\s+(?<host>http(?:s|)://.*)\s*;~`

**TL;DR**:

 * Use this regex if we don't want to sanitize input and be able to list all proxies, even if they are broken: `~proxy_pass\s+(?<host>.*)\s*;~` (at leasts makes it easier/possible to see proxies to try and remove them with `valet unproxy`)
 * Use this regex if we plan to sanitize input to make sure the proxy host used looks like a valid URL: `~proxy_pass\s+(?<host>http(?:s|)://.*)\s*;~`

Reply:
I'm unable to recall how I arrived at using that regex. I share your reasoning.

One adjustment I'd likely make to your 2nd proposed regex:

```diff
-~proxy_pass\s+(?<host>http(?:s|)://.*)\s*;~
+~proxy_pass\s+(?<host>https?://.*)\s*;~
```
(the `(?:s|)` can just be `s?`, unless I'm misreading your intention)

Reply:
@simensen I've updated the PR using your feedback. Thanks!

Reply:
Yeah, that looks good. :) Thanks!